### PR TITLE
Fixes Dreamchecker warnings in the codebase

### PR DESCRIPTION
**Pull request by SteelSlayer**
## What Does This PR Do
Fixes Dreamchecker warnings in the codebase.

There was a warning with one of the procs in `TextFilters.dm`, but nothing in that file was used anyway, so I just deleted it.

## Changelog
🆑
fix: Fixes 'moderate' level hallucinations not occuring
/🆑

### CODE REVIEW

**Comment by Sirryan2002** [comment]
It looks like everything in text filters was imported like 10 years ago and was never used

File: code/modules/mob/mob_helpers.dm [hunk 1]
```diff
@@ -221,9 +221,8 @@
 			if(1,3,5,8)	newletter="[lowertext(newletter)]"
 			if(2,4,6,15)	newletter="[uppertext(newletter)]"
 			if(7)	newletter+=pick(slurletters)
```
Comment (farie82) [inline code/modules/mob/mob_helpers.dm:223]: Ah yes why use logical X to Y when you can put random numbers in here.
Either somebody was slurring themselves when they wrote this. Or they are trying to counteract the bias of byonds random number generator.

File: code/modules/power/singularity/singularity.dm [hunk 2]
```diff
@@ -335,10 +335,10 @@
 	var/dir2 = 0
 	var/dir3 = 0
 	switch(direction)
-		if(NORTH||SOUTH)
+		if(NORTH, SOUTH)
```
Comment (farie82) [inline code/modules/power/singularity/singularity.dm:338]: These issues might actually be why the singulo sometimes goes through containment...

File: code/modules/reagents/chemistry/reagents/toxins.dm [hunk 3]
```diff
@@ -887,12 +887,11 @@
...
 	if(!M.nutrition)
-		switch(rand(1,3))
-			if(1)
-				to_chat(M, "<span class='warning'>You feel hungry...</span>")
-			if(2)
-				update_flags |= M.adjustToxLoss(1, FALSE)
-				to_chat(M, "<span class='warning'>Your stomach grumbles painfully!</span>")
+		if(prob(33))
+			to_chat(M, "<span class='warning'>You feel hungry...</span>")
+		else if(prob(33))
```
Comment (farie82) [inline code/modules/reagents/chemistry/reagents/toxins.dm:892]: Might be my morning brain. But this won't lead to the exact odds as before.
```suggestion
		else if(prob(50))
```
50% chance on a 66% to get into this step will result in a 33% chance to get this result.
So each option here has a 33% chance of happening in the end.

File: code/modules/reagents/chemistry/reagents/alcohol.dm [hunk 4]
```diff
@@ -1438,12 +1438,11 @@
...
 	if(!M.nutrition)
-		switch(rand(1, 3))
-			if(1)
-				to_chat(M, "<span class='warning'>You feel hungry...</span>")
-			if(2)
-				update_flags |= M.adjustToxLoss(1, FALSE)
-				to_chat(M, "<span class='warning'>Your stomach grumbles painfully!</span>")
+		if(prob(33))
+			to_chat(M, "<span class='warning'>You feel hungry...</span>")
+		else if(prob(33))
```
Comment (farie82) [inline code/modules/reagents/chemistry/reagents/alcohol.dm:1443]: Same here
```suggestion
		else if(prob(50))
```

File: code/modules/hallucinations/hallucinations.dm [hunk 5]
```diff
@@ -56,7 +56,7 @@ GLOBAL_LIST_INIT(hallucinations, list(
 	switch(rand(100))
 		if(0 to HALLUCINATE_MINOR_WEIGHT)
 			severity = HALLUCINATE_MINOR
-		if((HALLUCINATE_MINOR_WEIGHT + 1) to HALLUCINATE_MODERATE_WEIGHT)
+		if((HALLUCINATE_MINOR_WEIGHT + 1) to (HALLUCINATE_MINOR_WEIGHT + HALLUCINATE_MODERATE_WEIGHT))
```
Comment (farie82) [inline code/modules/hallucinations/hallucinations.dm:59]: This should fix moderate hallucinations not occurring. Unsure if this happens on live.
If so, can you add this to the CL?

File: code/__HELPERS/type2type.dm [hunk 6]
```diff
@@ -385,9 +385,9 @@
 		switch(child)
 			if(/datum)
 				return null
-			if(/obj || /mob)
+			if(/obj, /mob)
```
Comment (farie82) [inline code/__HELPERS/type2type.dm:388]: Seems like only components use this for their logic. Don't think stuff will break or change due to how our components are set up though.

**Review by farie82 — CHANGES_REQUESTED** [review]
Two minor things. Looks good otherwise!
The switch if changes `X || Y` -> `X, Y` actually change the logic being applied. This should fix bugs though and not create them

File: code/modules/reagents/chemistry/reagents/toxins.dm (hunk 3 above)
Comment (SteelSlayer) [reply code/modules/reagents/chemistry/reagents/toxins.dm:892]: Hm, this doesn't make sense to me 😅 

My thought process here is that `rand(1, 3)` gives us a ~33% chance of getting each of the numbers 1, 2 and 3. With old code there is a ~66% chance of something happening, and ~33% of something not happening.

So I just took the 66% and split it up into two 33%'s.

File: code/modules/hallucinations/hallucinations.dm (hunk 5 above)
Comment (SteelSlayer) [reply code/modules/hallucinations/hallucinations.dm:59]: Yeah, will add it

File: code/modules/mob/mob_helpers.dm (hunk 1 above)
Comment (SteelSlayer) [reply code/modules/mob/mob_helpers.dm:223]: Yeah, god this is so archaic. I don't know much about the bias of byond's RNG system though.

File: code/modules/reagents/chemistry/reagents/toxins.dm (hunk 3 above)
Comment (farie82) [reply code/modules/reagents/chemistry/reagents/toxins.dm:892]: The no op looks a bit wack to me. Wackier than a switch with not all cases covered.
Unsure what maints think of it though.

For the probability stuff, you can try to imagine 100 coins. If you want exactly 33 of them (1/3 rounded down) you will take 1/3 of it (`prob(33)`).
Then the remainder is 67. Taking half of that will be 33 (rounded down) so `prob(50)` in our case.

I however do agree that this is less intuitive at first glance...

Comment (AffectedArc07) [reply code/modules/reagents/chemistry/reagents/toxins.dm:892]: use `pass()` if you need a no op

**Review by farie82 — APPROVED** [review]

File: code/game/machinery/syndicatebeacon.dm [hunk 7]
```diff
@@ -49,12 +49,11 @@
...
-		switch(rand(1,2))
-			if(1)
-				temptext = "<font color=red><i><b>Double-crosser. You planned to betray us from the start. Allow us to repay the favor in kind.</b></i></font>"
-				src.updateUsrDialog()
-				spawn(rand(50,200)) selfdestruct()
-				return
+		if(prob(50))
+			temptext = "<font color=red><i><b>Double-crosser. You planned to betray us from the start. Allow us to repay the favor in kind.</b></i></font>"
+			updateUsrDialog()
+			spawn(rand(50,200)) selfdestruct()
```
Comment (AffectedArc07) [inline code/game/machinery/syndicatebeacon.dm:55]: Any chance you can make this a timer at the same time? 

`spawn()` needs to be shot.

**Review by AffectedArc07 — APPROVED** [review]
All good apart from that other remark

File: code/game/machinery/syndicatebeacon.dm (hunk 7 above)
Comment (SteelSlayer) [reply code/game/machinery/syndicatebeacon.dm:55]: Yeah I can do that.

**Review by dearmochi — APPROVED** [review]

**Pull request merged**